### PR TITLE
Avoid stopping SuSEfirewall2 for SLE15

### PR DESCRIPTION
**Pull request by asmorodskyi**

### CODE REVIEW

**Comment by asmorodskyi** [comment]
proof run http://kimball.arch.suse.de/tests/296

**Comment by okurz** [comment]
Why should we avoid that? I assume you want to fix something? Can you reference a ticket or a failing job run? Maybe the title is just misleading and you want to rather adjust the test to support both SLE 12 as well as SLE 15 and on SLE 15 use firewallctl ?

**Comment by asmorodskyi** [comment]
failing job https://openqa.suse.de/tests/1265210. For me I think is too much to create ticket in this case I will treat this as bureaucracy, but if you insist I will create one. As far as I know there is no SuSeFirewall in SLE15 anymore

**Comment by okurz** [comment]
No, I did not say that you have to create a new ticket.

Yes, there is no SUSEFirewall2 anymore in SLE but that doesn't mean there is *no* firewall -> firewalld

**Comment by asmorodskyi** [comment]
probably at some point I would drop this line at all because I am not sure if we actually need to do this . For now it's working without stopping firewalld I have proof run which proof it. Isn't this enough ?

**Comment by okurz** [comment]
No, sorry. does not make sense. Maybe @rwx788 and @SergioAtSUSE can help here? They take a bit more care about firewall related changes

File: tests/console/wicked_before_test.pm [hunk 1]
```diff
@@ -13,13 +13,13 @@
 use base "consoletest";
 use strict;
 use testapi;
-use utils qw(systemctl setup_static_network);
+use utils;
```
Comment (SergioAtSUSE) [inline tests/console/wicked_before_test.pm]: Why? using *qw(systemctl setup_static_network)* should be more efficient. @foursixnine knows.

File: tests/console/wicked_before_test.pm [hunk 2]
```diff
@@ -13,13 +13,13 @@
...
 use mm_network;
 
 sub run {
     my ($self) = @_;
     select_console('root-console');
-    assert_script_run "rcSuSEfirewall2 stop";
+    assert_script_run "rcSuSEfirewall2 stop" unless is_sle() && sle_version_at_least('15');
```
Comment (SergioAtSUSE) [inline tests/console/wicked_before_test.pm]: Then, shouldn't you stop firewalld ?

Comment (asmorodskyi) [reply tests/console/wicked_before_test.pm]: I have a proof run which shows that test is working without this https://openqa.suse.de/tests/1265210. Give me good reason why I should do this ?

**Comment by asmorodskyi** [comment]
@rwx788 , @okurz  it is networking testing so I am stopping firewall because this test not meant to test firewall but network. 
1. I have this run which showing that everything is working when I am stopping "SuSEfirewall2" on Tumbleweed https://openqa.opensuse.org/tests/540291 
2. I have this run which showing that it's not working with stopping "SuSEfirewall2" on SLE15 https://openqa.suse.de/tests/1269117
3. I have this run which showing that it's working when I just skip this line for SLE15  https://openqa.suse.de/tests/1265210

So I don't want change something for Tumbleweed (as it is already working ) and I want to fix it with minimal changes for SLE 15

File: tests/console/wicked_before_test.pm (hunk 1 above)
Comment (okurz) [reply tests/console/wicked_before_test.pm]: No, I removed that sleep call with e0aea3ce already :)

If you find any more then please
* remove them
* don't accept PRs adding any

It's not (only) about performance but about knowing explicitly what functions are used from where

**Comment by rwx788** [comment]
@asmorodskyi that's not what me and @SergioAtSUSE are saying here. We were disabling firewall. Now in sle 15 we get firewalld, so should still stop this service. The reason why test work may be that firewall doesn't work properly yet. But if you're saying that firewalld service is not configured and stop command will fail, put the comment in the code and we can merge it and adjust later. If disabling firewall on sle15 is not required, maybe it's not required anywhere, so let's remove this step completely then. But I guess it's not what we need here.

File: tests/console/wicked_before_test.pm (hunk 1 above)
Comment (asmorodskyi) [reply tests/console/wicked_before_test.pm]: In any case automation test should not care about each millisecond for sure . It is not unit tests , our tests approximate run is  30 minutes  or even more. Why we should care about few milliseconds which no one measure BTW ?
Can you clarify use case   for " knowing explicitly what functions are used from where" ? I mean if function is already mentioned in file you can just use it , if not you anyway need search globally to find a file where it is declared. So how you can use info that is_sle is declared in utils ?

**Comment by asmorodskyi** [comment]
exactly it's not what we need here. I am maintainer of this test case and I made decision to fix it in this way. I don't see point in such a big flame around such small thing

**Comment by asmorodskyi** [comment]
new proof run http://kimball.arch.suse.de/tests/304 ....

File: tests/console/wicked_before_test.pm [hunk 3]
```diff
@@ -13,13 +13,18 @@
...
 sub run {
     my ($self) = @_;
     select_console('root-console');
-    assert_script_run "rcSuSEfirewall2 stop";
+    if (is_sle() && sle_version_at_least('15')) {
+        systemctl('stop firewalld');
+    }
+    else {
+        assert_script_run('rcSuSEfirewall2 stop');
+    }
```
Comment (okurz) [inline tests/console/wicked_before_test.pm]: The commands are actually equivalent, it's only a different name for the service so the following can also work

```perl
my $service = (is_sle && sle_version_at_least('15')) ? 'firewalld' : 'SuSEfirewall2';
systemctl "stop $service";
```

**Comment by asmorodskyi** [comment]
fixed PR message

**Comment by rwx788** [comment]
You've fixed commit message, please change PR message as well

**Pull request merged**